### PR TITLE
media-sound/mpdscribble update live build and QA fix

### PR DESCRIPTION
**Pull request by cstew2**
**Changes**
- Fixed the live ebuild to work with its github master branch
- Fixed QA issue regarding installing paths instead of creating them in the openrc script for live and stable versions

Closes: https://bugs.gentoo.org/605374
Signed-off-by: Christopher Steward <cstew764@gmail.com>

### CODE REVIEW

File: media-sound/mpdscribble/mpdscribble-0.22.ebuild [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 1999-2018 Gentoo Foundation
+# Copyright 1999-2019 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=6
```
Comment (ajakk) [inline media-sound/mpdscribble/mpdscribble-0.22.ebuild:4]: Can we update to EAPI=7 at this point?

**Review by ajakk — CHANGES_REQUESTED** [review]

File: media-sound/mpdscribble/mpdscribble-0.22.ebuild (hunk 1 above)
Comment (cstew2) [reply media-sound/mpdscribble/mpdscribble-0.22.ebuild]: I think so, looking at the docs I don't see any removed/banned things used in these ebuilds.

**Comment by scardracs** [comment]
You may want to revbump to r1 as you are directly modifying an already stable ebuild. Or to update to 0.23 https://www.musicpd.org/clients/mpdscribble/

**Comment by scardracs** [comment]
I suggest to you to revbump as 0.22 already exist or better to update to 0.23 https://www.musicpd.org/clients/mpdscribble/

**Comment by scardracs** [comment]
Be sure to revert modifications you have made to 0.22 otherwise it will continue to warn you

**Comment by cstew2** [comment]
Repoman seems to like to change the copyright date on it's own, will that affect the CI?

**Comment by scardracs** [comment]
> Repoman seems to like to change the copyright date on it's own, will that affect the CI?

Yeah, just use repoman -f commit (be sure to have fixed other complains before pushing)

**Comment by cstew2** [comment]
Even with the -f argument it seems to do it, I may have to commit without repoman if that is the issue.

**Comment by scardracs** [comment]
> Even with the -f argument it seems to do it, I may have to commit without repoman if that is the issue.

Run git rebase -i master and change fix with drop (be sure to copy 0.23 to another folder), copy your file/s back and run only git add for these files. This should fix your problem

**Comment by scardracs** [comment]
Probably you have to rebase because it's blocked in pending

**Comment by cstew2** [comment]
I did do an interactive rebase and dropped the commits to 0.23 then force pushed. Should I pull and then rebase again? It does look like the 0.22 ebuild was changed between when I opened this pull request and now.

**Comment by scardracs** [comment]
Now it looks good to me, @thesamesam @juippis what you think?

File: media-sound/mpdscribble/mpdscribble-9999.ebuild [hunk 2]
```diff
@@ -12,27 +12,19 @@ EGIT_REPO_URI="https://github.com/MusicPlayerDaemon/mpdscribble.git"
 LICENSE="GPL-2"
 SLOT="0"
 KEYWORDS=""
-IUSE="+curl"
+IUSE=""
```
Comment (scardracs) [inline media-sound/mpdscribble/mpdscribble-9999.ebuild:15]: When IUSE is empty you can remove it

File: media-sound/mpdscribble/mpdscribble-0.23.ebuild [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
+IUSE=""
+
+RDEPEND=">=media-libs/libmpdclient-2.5
+	>=dev-libs/boost-1.62
+	dev-libs/libgcrypt
+	>=net-misc/curl-7.18"
```
Comment (scardracs) [inline media-sound/mpdscribble/mpdscribble-0.23.ebuild:20]: Dependencies are alphabetically ordered, please fix

File: media-sound/mpdscribble/mpdscribble-9999.ebuild [hunk 4]
```diff
@@ -12,27 +12,19 @@ EGIT_REPO_URI="https://github.com/MusicPlayerDaemon/mpdscribble.git"
...
-}
-
-src_configure() {
-	econf \
-		--with-http-client=$(usex curl curl soup)
-}
+RDEPEND=">=media-libs/libmpdclient-2.5
+	>=dev-libs/boost-1.62
+	dev-libs/libgcrypt
+	>=net-misc/curl-7.18"
```
Comment (scardracs) [inline media-sound/mpdscribble/mpdscribble-9999.ebuild:20]: Same as before

**Comment by scardracs** [comment]
Sorry if I haven't noticed it before, I was too concentrated on help you out with broken ci. You can make these 3 things and rebase :)

File: media-sound/mpdscribble/mpdscribble-0.23.ebuild (hunk 3 above)
Comment (scardracs) [reply media-sound/mpdscribble/mpdscribble-0.23.ebuild:20]: You can also change them in:

dev-libs/boost
dev-libs/libgcrypt
media-libs/libmpdclient
net-misc/curl

as these blockers doesn't exist anymore

File: media-sound/mpdscribble/mpdscribble-9999.ebuild [hunk 5]
```diff
@@ -12,27 +12,19 @@ EGIT_REPO_URI="https://github.com/MusicPlayerDaemon/mpdscribble.git"
...
+DEPEND="${RDEPEND}"
 
 src_install() {
 	default
+	meson_src_install
+	rm -r "${D}/usr/share/doc/mpdscribble"
```
Comment (scardracs) [inline media-sound/mpdscribble/mpdscribble-9999.ebuild:26]: rm -r "${D}/usr/share/doc/mpdscribble" || die

File: media-sound/mpdscribble/mpdscribble-0.23.ebuild [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+DEPEND="${RDEPEND}"
+
+src_install() {
+	default
+	meson_src_install
+	rm -r "${D}/usr/share/doc/mpdscribble"
```
Comment (scardracs) [inline media-sound/mpdscribble/mpdscribble-0.23.ebuild:26]: External commands need || die

rm -r "${D}/usr/share/doc/mpdscribble" || die

**Comment by scardracs** [comment]
Sorry if I haven't noticed them before, I was too concentrated on help you out with broken ci. You can make these changes and rebase :)

File: media-sound/mpdscribble/mpdscribble-0.23.ebuild [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+EAPI=7
+
+inherit meson ninja-utils
+
+DESCRIPTION="An MPD client that submits information to Audioscrobbler"
+HOMEPAGE="https://www.musicpd.org/clients/mpdscribble/"
+SRC_URI="https://www.musicpd.org/download/${PN}/${PV}/${P}.tar.xz"
+
+LICENSE="GPL-2"
```
Comment (00-matt) [inline media-sound/mpdscribble/mpdscribble-0.23.ebuild:12]: I think this should be GPL-2+, from the header of `Main.cxx`:

```
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
```

File: media-sound/mpdscribble/mpdscribble-9999.ebuild [hunk 8]
```diff
@@ -12,27 +12,19 @@ EGIT_REPO_URI="https://github.com/MusicPlayerDaemon/mpdscribble.git"
 LICENSE="GPL-2"
```
Comment (00-matt) [inline media-sound/mpdscribble/mpdscribble-9999.ebuild:12]: Same as above (`GPL-2+`).

**Review by 00-matt — COMMENTED** [review]
LICENSE nitpick :)

**Comment by ajakk** [comment]
This was superceded by: https://github.com/gentoo/gentoo/pull/22254

**Pull request closed without merging**